### PR TITLE
store: do not modify key during scanning

### PR DESCRIPTION
**Pull request by xiang90**
Fix https://github.com/coreos/etcd/issues/6328

### CODE REVIEW

**Comment by heyitsanthony** [comment]
so it was adding a trailing slash and that messed up the prev node matching in the code below? lgtm

**Comment by xiang90** [comment]
@heyitsanthony 

It is also in a for loop. The scan in any following cycles of the loop will be "messed up" too.

File: store/event_history.go [hunk 1]
```diff
@@ -83,12 +83,12 @@ func (eh *EventHistory) scan(key string, recursive bool, index uint64) (*Event,
 
 			if recursive {
 				// add tailing slash
-				key = path.Clean(key)
-				if key[len(key)-1] != '/' {
-					key = key + "/"
+				nkey := path.Clean(key)
```
Comment (heyitsanthony) [inline store/event_history.go]: looking at this again, I think this only needs to be computed once, before the for loop?

Comment (xiang90) [inline store/event_history.go]: yes, but i do not really want to touch/change v2 related code in any significant way... so if this works, then great...

Comment (heyitsanthony) [inline store/event_history.go]: ok, fair enough

**Pull request merged**